hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicationTask.java

```diff
@@ -181,19 +223,38 @@ public Boolean call() {
       List<RowMutations> rowMutationsList = new ArrayList<>(cellsToReplicateByRow.size());
       for (Map.Entry<ByteRange, List<Cell>> cellsByRow : cellsToReplicateByRow.entrySet()) {
         // Create a rowMutations and add it to the list to be flushed to CBT.
-        RowMutations rowMutations =
-            buildRowMutations(cellsByRow.getKey().deepCopyToNewArray(), cellsByRow.getValue());
-        rowMutationsList.add(rowMutations);
+
+        // TODO - added for debugging REMOVE
+        /**
+         * Iterator<Cell> cellIt = cellsByRow.getValue().iterator(); while (cellIt.hasNext()) { Put
+         * put = new Put(cellsByRow.getKey().deepCopyToNewArray());
+         *
+         * <p>for (int i = 0; i < (100_000-1) && cellIt.hasNext(); i++) { put.add(cellIt.next()); }
+         *
+         * <p>table.put(put); }
+         */
+        List<RowMutations> rowMutations =
```

Comment:
Doing single rowmutation in a batch is very inefficient. This is going to build replication lag as we are sending an RPC per put.